dlt/common/destination/reference.py

```diff
@@ -498,6 +503,30 @@ def should_truncate_table_before_load_on_staging_destination(self, table: TTable
         return True
 
 
+class SupportsDataAccess(ABC):
+    """Add support for accessing data as arrow tables or pandas dataframes"""
+
+    @abstractmethod
+    def iter_df(
```

Comment:
I wouldn't explicitly expose Pandas dataframes. I would only expose Arrow data structures , because the user can call `to_pandas` on those.

Reply:
@jorritsandbrink there are some destinations that have native pandas support and I think it would be cool to be able to expose those directly for the user